### DISCUSSION TITLE
Restore the best model

### DISCUSSION THREAD

**Discussion by mpariente** [Lightning-AI/pytorch-lightning#5812] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
What would be the most lightning way to restore the best model? Either directly after training (in the same script) or for later use (in another script)? 

Thanks in advance !

**Comment by neggert** [comment, marked as answer]
You can use the checkpont callback to save the only the best model as described here: https://williamfalcon.github.io/pytorch-lightning/Trainer/Checkpointing/ (note that that doc needs to be updated. Use `save_top_k=1` instead of `save_best_only`)

You can then use the `load_from_checkpoint` method to restore your checkpoint: https://williamfalcon.github.io/pytorch-lightning/LightningModule/methods/#load_from_metrics

**Comment by Borda** [comment]
in the last version, there is also `Trainer` initialization parameter `resume_from_checkpoint`

**Comment by mpariente** [comment]
> You can use the checkpont callback to save the only the best model as described here: https://williamfalcon.github.io/pytorch-lightning/Trainer/Checkpointing/ (note that that doc needs to be updated. Use `save_top_k=1` instead of `save_best_only`)

I didn't pay attention, I was using `save_best_only`, thanks !

Thank you both for you answers, but what if I want to keep my checkpoints, and still restore the best one automatically, how would I do?

**Comment by neggert** [comment]
Right now, I don't think the score is saved with the checkpoint. You'd need to take a look at your logs and manually select the best one.
